### PR TITLE
Adding the sensors for goal conditioning

### PR DESCRIPTION
**Pull request by vincentpierre**
Goal conditioning tensor modifications for v1

### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Sensors/VectorSensorComponent.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            return m_sensor;
+        }
+
+
+        /// <summary>
+        /// Update fields that are safe to change on the Sensor at runtime.
+        /// </summary>
+        internal void UpdateSensor()
+        {
+            if (m_sensor != null)
```
Comment (awjuliani) [inline com.unity.ml-agents/Runtime/Sensors/VectorSensorComponent.cs:56]: Since we don't update any fields during runtime, do we need this function here?

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Sensors/VectorSensorComponent.cs:56]: No, don't think so.

File: com.unity.ml-agents/Runtime/Sensors/VectorSensorComponent.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Unity.MLAgents.Sensors
+{
+    [AddComponentMenu("ML Agents/Vector Sensor", (int)MenuGroup.Sensors)]
+    public class VectorSensorComponent : SensorComponent
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensors/VectorSensorComponent.cs:7]: Can you add a name field here too, and pass to the VectorSensor constructor?

File: com.unity.ml-agents/Runtime/Sensors/VectorSensor.cs [hunk 3]
```diff
@@ -8,20 +8,21 @@ namespace Unity.MLAgents.Sensors
...
         string m_Name;
+        ObservationType m_ObservationType;
 
         /// <summary>
         /// Initializes the sensor.
         /// </summary>
         /// <param name="observationSize">Number of vector observations.</param>
         /// <param name="name">Name of the sensor.</param>
-        public VectorSensor(int observationSize, string name = null)
+        public VectorSensor(int observationSize, string name = null, ObservationType observationType = ObservationType.Default)
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensors/VectorSensor.cs:25]: Maybe if `observationType != ObservationType.Default` we should add it to the default name value too?

**Pull request closed without merging**